vcd/resource_vcd_nsxt_edgegateway_dns.go

```diff
@@ -0,0 +1,373 @@
+			// removed from state
+			d.SetId("")
+		}
+		return diag.Errorf("[edge gateway dns read] error retrieving NSX-T Edge Gateway DNS config: %s", err)
+	}
+
+	dns, err := nsxtEdge.GetDnsConfig()
+	if err != nil {
+		return diag.Errorf("[edge gateway dns read] error retrieving NSX-T Edge Gateway DNS config: %s", err)
+	}
+	d.SetId(dns.EdgeGatewayId)
+	dSet(d, "edge_gateway_id", dns.EdgeGatewayId)
+
+	err = setNsxtEdgeGatewayDnsConfig(d, dns.NsxtEdgeGatewayDns)
+	if err != nil {
+		return diag.Errorf("[edge gateway dns read] error storing state: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdNsxtEdgegatewayDnsDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	unlock, err := vcdClient.lockParentVdcGroupOrEdgeGateway(d)
+	if err != nil {
+		return diag.Errorf("[edge gateway dns delete] %s", err)
+	}
+	defer unlock()
+
+	orgName := d.Get("org").(string)
+	edgeGatewayId := d.Get("edge_gateway_id").(string)
+
+	nsxtEdge, err := vcdClient.GetNsxtEdgeGatewayById(orgName, edgeGatewayId)
+	if err != nil {
+		return diag.Errorf("[edge gateway dns delete] error retrieving Edge Gateway: %s", err)
+	}
+
+	dns, err := nsxtEdge.GetDnsConfig()
+	if err != nil {
+		return diag.Errorf("[edge gateway dns delete] error retrieving DNS Configuration: %s", err)
+	}
+
+	err = dns.Delete()
+	if err != nil {
+		return diag.Errorf("[edge gateway dns delete] error deleting DNS Configuration: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdNsxtEdgegatewayDnsImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] NSX-T Edge Gateway DNS import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 3 {
+		return nil, fmt.Errorf("resource name must be specified as org-name.vdc-name.nsxt-edge-gw-name or org-name.vdc-group-name.nsxt-edge-gw-name")
+	}
+	orgName, vdcOrVdcGroupName, edgeName := resourceURI[0], resourceURI[1], resourceURI[2]
+
+	vcdClient := meta.(*VCDClient)
+	vdcOrVdcGroup, err := lookupVdcOrVdcGroup(vcdClient, orgName, vdcOrVdcGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	if !vdcOrVdcGroup.IsNsxt() {
+		return nil, fmt.Errorf("this resource is only supported on NSX-T backed Edge Gateways")
+	}
+
+	edge, err := vdcOrVdcGroup.GetNsxtEdgeGatewayByName(edgeName)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve NSX-T Edge Gateway with ID '%s': %s", d.Id(), err)
+	}
+
+	dSet(d, "org", orgName)
+	dSet(d, "edge_gateway_id", edge.EdgeGateway.ID)
+	// Storing Edge Gateway ID and Read will retrieve all other data
+	d.SetId(edge.EdgeGateway.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func getNsxtEdgeGatewayDnsConfig(d *schema.ResourceData, vcdClient *VCDClient) (*types.NsxtEdgeGatewayDns, error) {
+	enabled := d.Get("enabled").(bool)
+	listenerIp := d.Get("listener_ip").(string)
+	snatRuleEnabled := d.Get("snat_rule_enabled").(bool)
```

Comment:
This field is a computed variable in the resource schema. Reading it in the `create` phase will set `false` (default boolean value), but I see the rule is created by default. 

Can the field `snat_rule_enabled` be user-modified (can a user prevent creating this rule)? If this is a read-only field - then I guess we shouldn't set it to the payload.

Reply:
You're right, updated now, even though as I replied in the lower comment, it doesn't make a difference.